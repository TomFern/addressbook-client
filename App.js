// Mobile Client example for semaphore-demo-cicd-kubernetes (addressbook server)

import React from 'react';
import { FlatList, ActivityIndicator, Text, View, StyleSheet } from 'react-native';

const API_SERVER = process.env.API_SERVER;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 40,
    marginHorizontal: 10,
    fontSize: 32,
  },
  item: {
    backgroundColor: '#f9c2ff',
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 8,
  },
  personName: {
    fontSize: 24,
  },
  importantText: {
      color: 'red',
      fontSize: 32,
  },
});

function Item({ firstName, lastName }) {
  return (
    <View style={styles.item}>
      <Text style={styles.personName}>{lastName}, {firstName}</Text>
    </View>
  );
}

export default class FetchAddressbook extends React.Component {
  constructor(props){
    super(props);
      this.state = { isLoading: true, nodata: true };
  }

  // fetch persons
  componentDidMount(){

    return fetch(API_SERVER + '/all')
      .then((response) => response.json())
      .then((responseJson) => {
        this.setState({
          isLoading: false,
          nodata: false,
          dataSource: responseJson,
        },
        function(){});
      })
      .catch((error) => {
        this.setState({
          isLoading: false,
          error: error,
        });
      });

  }

  render(){

    if (this.state.isLoading){
      return (
        <View style={styles.container}>
          <ActivityIndicator/>
        </View>
      );
    }

    if (this.state.nodata) {
      return (
        <View style={styles.container}>
            <Text style={styles.importantText}>No data found.</Text>
        </View>
      );
    }

    return (
        <View style={styles.container}>
            <Text style={styles.personName}>Addressbook</Text>
            <FlatList
                data={this.state.dataSource}
                renderItem={({ item }) => <Item lastName={item.lastName} firstName={item.firstName} />}
                keyExtractor={(item) => item.id.toString()}
            />
        </View>
    );

  }
}

