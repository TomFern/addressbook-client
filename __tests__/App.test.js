import ReactNative from 'react-native';
// import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import React from 'react';
import App from '../App';

const {View, Text, Switch} = ReactNative;

describe('jest snapshot tests', () => {

  beforeEach(() => {
    fetch.resetMocks()
  });

  it('renders correctly', () => {
      // TODO: invalid-json logged
    // fetch.mockResponseOnce([{
    //     id: 1, 
    //     firstName: "Foo",
    //     lastName: "Bar"
    // }]);
 fetch.mockResponseOnce();
 // fetch.mockResponseOnce(JSON.stringify({}));
 // fetch.mockResponseOnce(JSON.stringify([{ id: 1 }]));
  // const onResponse = jest.fn();
  // const onError = jest.fn();
      // fetch.mockResponseOnce('[{"id":1,"firstName":"Les","lastName":"Claypool","createdAt":"2019-12-23T15:08:47.991Z","updatedAt":"2019-12-23T15:08:47.991Z"},{"id":2,"firstName":"Al","lastName":"Pacino","createdAt":"2019-12-23T15:08:48.041Z","updatedAt":"2019-12-23T15:08:48.041Z"}]');
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
