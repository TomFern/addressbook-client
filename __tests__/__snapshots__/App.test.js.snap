// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest snapshot tests renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "fontSize": 32,
      "marginHorizontal": 10,
      "marginTop": 40,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
